syncer/syncer_test.go

```diff
@@ -634,27 +569,6 @@ func TestSync_AlsoSyncProcessedLayer(t *testing.T) {
 	require.Equal(t, lyr, ts.syncer.getLastSyncedLayer())
 }
 
-func TestSyncer_setSyncedTwice_NoError(t *testing.T) {
-	ts := newSyncerWithoutSyncTimer(t)
-
-	sync := ts.syncer.RegisterForSynced(context.TODO())
-	select {
-	case <-sync:
-		require.Fail(t, "should not have reached synced state yet")
-	case <-time.After(100 * time.Millisecond):
-	}
-
-	ts.syncer.setSyncState(context.TODO(), synced)
-
-	select {
-	case <-sync:
-	case <-time.After(1 * time.Second):
-		require.Fail(t, "should have reached synced state")
-	}
-
-	require.NotPanics(t, func() { ts.syncer.setSyncState(context.TODO(), synced) })
-}
```

Comment:
How do we deal with regressions? I wrote this test when there was a bug in the code that caused a panic. This test 1) verifies that the bug was fixed and 2) prevents a regression.

Reply:
this doesn't quite make sense without notifying channels. it's basically setting variables. having this test would be more confusing.

Reply:
This is not true. This bug occurred in a system test.

Here are the exact circumstances:

1. A go routine called `RegisterForSynced` and received a channel that it was using to be notified when the synced state happened.
2. A second go routine called `setSyncState` with `synced` triggering the notification of the first go routine.
3. The second go routine later called `setSyncState` again, without someone Registering for it.
4. PANIC -> because `setSyncState` tried to close channels that ware already closed

The fix was clearing the internal map holding the channels created by `RegisterForSynced` after closing them.

Steps I took to fix the bug:

1. Write a test that should pass but fails because of the bug
2. Fix the bug
3. Verify that the test is now passing and leave it to avoid a regression

Reply:
i am confused. i did NOT TestSyncer_setATXSyncedTwice_NoError.
i only removed TestSyncer_setSyncedTwice_NoError because there is no longer a internal map holding the channels for synced-ness, only for ATX synced-ness.